nextcord/voice_recording/opus.py

```diff
@@ -0,0 +1,65 @@
+# SPDX-License-Identifier: MIT
+
+from __future__ import annotations
+
+import gc
+import logging
+from threading import Event, Thread
+from time import sleep
+from typing import TYPE_CHECKING, Dict
+
+import nextcord as nc
+
+if TYPE_CHECKING:
+    from .core import OpusFrame, RecorderClient
+
+
+class DecoderThread(Thread, nc.opus._OpusStruct):
+    def __init__(self, recorder) -> None:
+        super().__init__(daemon=True)
+
+        self.recorder: RecorderClient = recorder
+        self.decode_queue: list[OpusFrame] = []
+        self.decoder: Dict[int, nc.opus.Decoder] = {}
+        self._end = Event()
+
+    def start(self) -> None:
+        self._end = Event()
+        super().start()
+
+    def decode(self, opus_frame) -> None:
+        self.decode_queue.append(opus_frame)
+
+    def run(self) -> None:
+        while not self._end.is_set():
+            try:
+                opus_frame: OpusFrame = self.decode_queue.pop(0)
+
+            except IndexError:
+                sleep(0.001)
```

Comment:
What and why?

Reply:
It is a while loop after all, sleeping when waiting for an item in the queue since the queue isn't expected to receive items more than that often. A frame only arrives every 20ms per user, sleeping during the wait can lower CPU usage.

Reply:
Also, the most of opus.py is copied from the previous PR

Reply:
Although there is probably a more elegant approach to waiting for an item in queue

Reply:
make the decode_queue an actual `queue.Queue` and use it's methods